### PR TITLE
feat(broker): open message subscription

### LINKED ISSUE #1014: Open a message subscription when an intermediate message catch event is entered
AT:
* when an intermediate message catch event is entered then a new message subscription is opened
* the message subscription is opened on one partition of the workflow instance topic
* the partition is chosen depending on a hash function which is applied to the extracted payload value of the correlation-key (i.e. `partition-id = partitions[hash(correlation-key(payload)) % partitions.lenth]`)
* the request to open the subscription is send over the message endpoint as "remote command" 
  * single message instead of response

Context:
* prefer single message over request-response to open the subscription to not block the workflow processing (both partitions can be on different nodes)

### PR DESCRIPTION
**Pull request by saig0**
* new SBE message to open a message subscription
* new record for message subscriptions
* send a command over the subscription endpoint when a message catch
event is entered
* write a message subscription record when a command is received

closes #1014

### CODE REVIEW

File: protocol/src/main/resources/protocol.xml [hunk 1]
```diff
@@ -40,6 +40,8 @@
         <validValue name="TOPIC">8</validValue>
         <validValue name="ID">9</validValue>
         <validValue name="MESSAGE">10</validValue>
+        <validValue name="MESSAGE_SUBSCRIPTION">11</validValue>
+        <validValue name="WORKFLOW_INSTANCE_SUBSCRIPTION">12</validValue>
```
Comment (ThorbenLindhauer) [inline protocol/src/main/resources/protocol.xml]: This is not used.

**Review by ThorbenLindhauer — COMMENTED** [review]
Test coverage:

* Two subscriptions with the same correlation key should go to the same partition
* This should work with and without a broker restart in between

I think these tests make particularly sense, because the current implementation assumes that the fetch topic response returns partition IDs in a consistent order. This dependency is not obvious and could lead to a regression in the future.

**Review by ThorbenLindhauer — CHANGES_REQUESTED** [review]
See comments above

**Comment by saig0** [comment]
I created a follow-up issue for the partition assignment test #1083 .

**Review by ThorbenLindhauer — APPROVED** [review]

**Pull request merged**